lib/hets/prove/evaluator.rb

```diff
@@ -13,17 +13,19 @@ class Evaluator
       # and
       # new(user, ontology, path: some_path, io: some_io)
       # where some_io needs to be an instance of IO or a Tempfile.
-      def initialize(user, ontology, version: nil, path: nil, io: nil)
+      def initialize(user, ontology, proof_attempts,
+                     version: nil, path: nil, io: nil)
         self.version = version
         self.path = path
         self.ontology = ontology
+        self.proof_attempts = proof_attempts
         self.user = user
         self.io = io
         initialize_handling
       end
 
       def import
-        callback = ProveEvaluator.new(self)
+        callback = ProveEvaluator.new(self, proof_attempts)
         ActiveRecord::Base.transaction requires_new: true do
           callback.process(:all, :start)
           parser.parse(callback: callback)
```

Comment:
I created #1295 for this and I want to finish that issue this week. The explanation of the classes for now:

* `Hets::Prove::Evaluator` is the entry point for evaluating the hets output. It calls the parser and passes the callbacks to the parser.
* `Hets::Prove::ProveEvaluator < Hets::BaseEvaluator` contains those callbacks and actually fills the `ProofAttempt` objects. It inherits from `BaseEvaluator` because the `BaseEvaluator` defines a convenient callback registration mechanism.